scripts/build.ps1

```diff
@@ -213,6 +213,19 @@ function Invoke-Build
     Write-Log "Invoke-Build: Complete. {$(Get-ElapsedTime($timer))}"
 }
 
+function Publish-PatchedDotnet { 
```

Comment:
I'm not 100% sure if that's the right thing to do. Can you explain in more detail why you need the file copying?

Reply:
This can be used to verify upfront that the integration with dotnet cli is not broken.

Reply:
Do you have any better ideas?

Reply:
@ViktorHofer could you address this one please? I am making a copy of the local dotnet to make sure we can detect that `dotnet test` is broken as part of our acceptance tests, instead of seeing it fail when we try to insert this dependency into the cli tooling. At the same time we don't want to fiddle with the local copy of dotnet, because that is used to do the actual build, and we don't want the developer to end up in a loop where changes to the code break the tool that builds the code. So making a copy, patching it and testing against that seemed like the best solution.

Reply:
OK got it, just wanted to understand in more detail what this is for. If this provides value then great :)

Reply:
how much does this increase the build times by though?

Reply:
Not dramatically, copying takes about a minute, and the added tests as well adding about 2 minutes to the overall run of 35 minutes. So it is a bit of a cost but we can now better test the `dotnet test` integration, but every new test that will test that will not get that performance hit.